### DISCUSSION TITLE
global ensemble TC genesis verification

### DISCUSSION THREAD

**Discussion by JiayiPeng-NOAA** [dtcenter/METplus#1304] [Statistical Computation] [answered]
Does the Met/MetPlus have the capability to verify GEFS (global ensemble) TC genesis ?

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @JiayiPeng-NOAA. MET does include a tool named TC-Gen which is described in this [Chapter](https://met.readthedocs.io/en/latest/Users_Guide/tc-gen.html?highlight=tc_gen) of the MET User's Guide. Here's an example call to tc_gen that's run as part of our automated tests:
```
tc_gen \
-genesis MET_test_input/tc_data/genesis/suite1/2016*/genesis*2016* \
-track MET_test_input/tc_data/genesis/atcf/2016 \
-config test/config/TCGenConfig_2016 \
-out MET_test_output/tc_gen/tc_gen_2016
```
And here's an example of some output contingency table counts that are produced:
```
LINE_TYPE TOTAL FY_OY FY_ON FN_OY FN_ON
CTC         789    10   493   286     0
```
So what are the required inputs for this?
- the '-genesis' option points to genesis "fort.66" file produced by the tracker. Here's one for example: [genesis.fort.66.suite1.2016111600.txt](https://github.com/dtcenter/METplus/files/7694770/genesis.fort.66.suite1.2016111600.txt)
 - the '-track' option points to a directory that contains BEST and operational track data. For example BDECK files along with operational ADECK files that contain CARQ track data.
 - the '-config' option is the config file. Here's the [config file](https://github.com/dtcenter/MET/blob/main_v10.0/test/config/TCGenConfig_2016) used for our sample test.

So if you have those genesis fort.66 files available for GEFS, then sure, you should be able to run TC-Gen with those inputs.

Hopefully that helps clarify.

**Comment by halperin-erau** [comment]
Hi @JiayiPeng-NOAA 

If you wish to verify probabilistic TC genesis forecasts, you can do so starting with the MET10.1.0-beta4 release.  To verify probabilistic forecasts with TC-Gen, your input data should follow a slightly modified ATCF e-deck format.  Here is an example of the format:

AL, 77, 2020082918, GN, GFSO, 120, 310N, 785W,  20,  120,  JHT, genFcst, 2020083106,  ,  ,  ,

"AL": Basin
"77": Arbitrary storm ID
"2020082918": Model initialization time
"GN": Genesis
"GFSO": Model identifier
"120": 120 h time period
"310N": Forecast genesis latitude
"785W": Forecast genesis longitude
"20": Forecast genesis probability for time period specified in next column
"120": 120 h forecast genesis window
"JHT": Forecast technique (can be any string)
"genFcst": Genesis forecast
"2020083106": Model forecast genesis valid time

Everything after the model forecast genesis valid time can be empty columns.  The "GN" and "genFcst" entries must be present for MET to recognize that the line refers to a TC genesis forecast.  The latitude, longitude, model initialization time, and model forecast genesis valid time also must be included for TC-Gen to conduct the verification.  Your input files must end with ".dat" and you should use the '-edeck' option when calling TC-Gen.
